### PR TITLE
fix: enable allOf transformation for parents without properties

### PR DESCRIPTION
**Pull request by andmatei**
## Description
`allOf` references to parents that have no properties or required properties should be removed as part of the `allOf` transformation.

Before:
```
    CreateEndpointRequest:
      type: object
      oneOf:
        - $ref: "#/components/schemas/CreateAWSEndpointRequest"
        - $ref: "#/components/schemas/CreateAzureEndpointRequest"
        - $ref: "#/components/schemas/CreateGCPEndpointGroupRequest"
    CreateAWSEndpointRequest:
      type: object
      allOf:
      - $ref: '#/components/schemas/ApiAtlasCreateEndpointRequestView'
      - type: object
         properties: ...
    CreateAzureEndpointRequest:
      type: object
      allOf:
      - $ref: '#/components/schemas/ApiAtlasCreateEndpointRequestView'
      - type: object
         properties: ...
    CreateGCPEndpointGroupRequest:
      type: object
      allOf:
      - $ref: '#/components/schemas/ApiAtlasCreateEndpointRequestView'
      - type: object
         properties: ...
```

After:
```
   CreateEndpointRequest:
      type: object
      oneOf:
        - $ref: "#/components/schemas/CreateAWSEndpointRequest"
        - $ref: "#/components/schemas/CreateAzureEndpointRequest"
        - $ref: "#/components/schemas/CreateGCPEndpointGroupRequest"
   CreateAWSEndpointRequest:
      type: object
      properties: ...
   CreateAzureEndpointRequest:
      type: object
      properties: ...
   CreateGCPEndpointGroupRequest:
      type: object
      properties: ...
```

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: tools/transformer/src/transformations/allOf.js [hunk 1]
```diff
@@ -28,7 +29,9 @@ function applyAllOfTransformations(api) {
   );
 
   for (let { path } of allOfTransformations) {
-    transformAllOf(path, api);
+    if(!isInlineModel(path)) {
```
Comment (andmatei) [inline tools/transformer/src/transformations/allOf.js:32]: Ignore inline models for `allOf` transformation as the parent will never have properties or be referenced by the children.

File: tools/transformer/src/engine/readers.js [hunk 2]
```diff
@@ -141,6 +142,13 @@ function getObjectFromYamlPath(path, obj) {
   return currObj;
 }
 
+function isInlineModel(path) {
+  const pathStack = path.split(".").reverse();
+  pathStack.pop();
+
+  return !(pathStack.pop() == "components" && pathStack.pop() == "schemas");
+}
```
Comment (matt-condon) [inline tools/transformer/src/engine/readers.js:150]: Is there any possibility of encountering a path with fewer than 2 segments?

**Review by matt-condon — DISMISSED** [review]
Nice one, very clean 🧹  

Just one minor flag

File: tools/transformer/src/engine/readers.js (hunk 2 above)
Comment (andmatei) [reply tools/transformer/src/engine/readers.js:150]: Under the `components` object, it should always have one more extra segment. But I will add some more validation for this.

**Comment by wtrocki** [comment]
There seems to be the issue with tools job trigger. It did not trigger. PR is missing transformated openapi file.

**Comment by andmatei** [comment]
For this version of the spec there are no changes in the transformed spec.

**Comment by wtrocki** [comment]
Looks like github glitch. Action is configured properly

https://github.com/mongodb/go-client-mongodb-atlas/blob/sdkv2/.github/workflows/tools.yaml#L5

**Comment by wtrocki** [comment]
> For this version of the spec there are no changes in the transformed spec.

Seen dev branch with changes. Looks very good.

**Pull request merged**